test/generate.test.ts

```diff
@@ -154,6 +154,98 @@ describe('generate', function () {
     });
   });
 
+  describe('scale-zero', () => {
+    it('should enumerate correct scale type when scale zero is used without bar mark or binning', () => {
+      const specQ = {
+        mark: Mark.POINT,
+        encodings: [
+          {
+            channel: Channel.X,
+            scale: {
+              zero: true,
+              type: {values: [undefined, ScaleType.SQRT, ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC]}
+            },
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+      const answerSet = generate(specQ, schema);
+      assert.equal(answerSet.length, 2);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+      assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.SQRT);
+    });
+
+    it('should enumerate correct scale properties with mark bar', () => {
+      const specQ = {
+        mark: Mark.BAR,
+        encodings: [
+          {
+            channel: Channel.X,
+            scale: {
+              zero: true,
+              type: {values: [undefined, ScaleType.SQRT, ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC]}
+            },
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+      const answerSet = generate(specQ, schema);
+      assert.equal(answerSet.length, 2);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+      assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.SQRT);
+    });
+
+    it('should enumerate correct scale properties with binned field and scale zero', () => {
+      const specQ = {
+        mark: Mark.POINT,
+        encodings: [
+          {
+            bin: true,
+            channel: Channel.X,
+            scale: {
+              zero: true,
+              type: {values: [undefined, ScaleType.SQRT, ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC]}
+            },
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+      const answerSet = generate(specQ, schema);
+
+      /* note for future developer:
+        You might expect this answerset to be completely empty since there is a constraint that prevents bin and zero from working together, however
+        if you look inside generate() you'll see that the checks for bin/zero only occur if they're enumerated. Since they're true/false values they don't run through
```

Comment:
"the checks for bin/zero only occur"  => "we only call <constraint name>, which requires Property.BIN and Property.SCALE_ZERO, if at least one of the properties is enumerated.  

"true/false" ==> I don't see any false here.  Just "true"

Can you be more careful with writing comments?  Otherwise your comments just add more confusion to readers.